pkg/services/preference/prefimpl/pref.go

```diff
@@ -214,7 +224,9 @@ func (s *Service) GetDefaults() *pref.Preference {
 		Timezone:        s.cfg.DateFormats.DefaultTimezone,
 		WeekStart:       s.cfg.DateFormats.DefaultWeekStart,
 		HomeDashboardID: 0,
-		JSONData:        &pref.PreferenceJSONData{},
+		JSONData: &pref.PreferenceJSONData{
+			Locale: s.cfg.DefaultLocale,
```

Comment:
since the index api is behind a feature flag, here probably needs to be behind the same feature toggle to make behaviour consistent,  s.Features.IsEnabled(featuremgmt.FlagInternationalization) then set default locale, otherwise the setting should be ignored?

Reply:
Default locale is now behind feature flag, and added the right tests for it.